lib/cancellable.ts

```diff
@@ -28,6 +28,12 @@ export class Cancellable {
     [inspect.custom]() {
         return "Cancellable {}";
     }
+
+    public static withTimeout(ms): Cancellable {
+        var cancel = new Cancellable();
+        setTimeout(() => cancel.cancel(), ms);
```

Comment:
```suggestion
        setTimeout(() => cancel.cancel(), ms).unref();
```

The `setTimeout()` will keep the Node.js event loop running. This means that short-lived CLI tools that don't explicitly `process.exit()` will end up sticking around until all pending `setTimeout()` callbacks have been run. Depending on the `ms` duration, this may take a while. One possible solution is to call `unref()` on the value returned by `setTimeout()`.